ivre/db/__init__.py

```diff
@@ -959,6 +962,33 @@ def __migrate_schema_hosts_10_11(doc):
                         pass
         return doc
 
+    @staticmethod
+    def __migrate_schema_hosts_11_12(doc):
+        """Converts a record from version 11 to version 12. Version 12 changes
+the structured output for fcrdns and rpcinfo script.
+
+        """
+        assert doc["schema_version"] == 11
+        doc["schema_version"] = 12
+        for port in doc.get('ports', []):
+            for script in port.get('scripts', []):
+                if script['id'] == "fcrdns":
+                    if "fcrdns" in script:
+                        # Since dots in keys are converted to
+                        # underscores, we need to convert them
+                        # back. It's OK to do so because underscores
+                        # cannot be used in FQDNs.
+                        script["fcrdns"] = [
+                            dict(result, name=name.replace('_', '.'))
```

Comment:
You might want to use your function `change_fcrdns`.

Reply:
`change_fcrdns` does not replace `_` by `.`.

Reply:
I missed that point. Nonetheless, maybe you should create another function rather than copy/paste the same logic in 3 different files ?

Reply:
Done!